src/Swift.Runtime/src/Swift/Runtime/ProtocolWitnessTable.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Runtime.InteropServices;
+
+namespace Swift.Runtime;
+
+public readonly struct ProtocolWitnessTable : IEquatable<ProtocolWitnessTable>
+{
+    private readonly IntPtr _handle;
+
+    private ProtocolWitnessTable(IntPtr handle)
+    {
+        _handle = handle;
+    }
+
+    public readonly static ProtocolWitnessTable Zero = default;
+
+    public bool IsValid => _handle != IntPtr.Zero;
+
+    public bool Equals(ProtocolWitnessTable other)
+    {
+        return _handle == other._handle;
+    }
+
+    public static ProtocolWitnessTable GetProtocolWitnessTable<TType, TProtocol>()
+        where TProtocol : class
+    {
+        var metadata = TypeMetadata.GetTypeMetadataOrThrow<TType>();
+        if (!ProtocolConformanceDescriptor.TryGetProtocolConformanceDescriptor<TType, TProtocol>(out var conformanceDescriptor))
+        {
+            throw new SwiftRuntimeException($"Unable to get protocol conformance descriptor for {typeof(TType)} and {typeof(TProtocol)}");
+        }
+
+        return GetProtocolWitnessTable(conformanceDescriptor.Value, metadata);
+    }
+
+    static ProtocolWitnessTable GetProtocolWitnessTable(ProtocolConformanceDescriptor conformanceDescriptor, TypeMetadata typeMetadata)
+    {
+        var witnessTable = swift_getWitnessTable(conformanceDescriptor, typeMetadata, IntPtr.Zero);
+        return witnessTable;
+    }
+
+    /// <summary>
+    /// Gets the protocol witness table for a given type and protocol.
+    /// </summary>
+    /// <param name="conformanceDescriptor">The protocol conformance descriptor.</param>
+    /// <param name="typeMetadata">The type metadata.</param>
+    /// <param name="instantiationArgs">The instantiation arguments used for conditional conformance.</param>
+    [DllImport(KnownLibraries.SwiftCore, CallingConvention = CallingConvention.Cdecl)]
```

Comment:
Do we want to use `CallingConvention.CallConvSwift`?

Reply:
I think this is a regular c-like call. So `Cdecl` is more appropriate. Anyway this call just accepts a bunch of pointers.

Reply:
The `cdecl` callconv will work here, but for consistency I suggest using Swift callconv. The libSwiftCore contains Swift standard library functions and structures.

Reply:
We only use `cdecl` in `Swift.Runtime`. I'm fine with changing all of them to `CallConvSwift`. Also should this be a `LibraryImport`?

Reply:
This function is exported from libSwiftCore with standard C calling convention: https://github.com/swiftlang/swift/blob/c1915c8cdd7052291f4761b37348f81511113a76/include/swift/Runtime/RuntimeFunctions.def#L1084 (`C_CC` means C calling convention).

So it is more appropriate to use Cdecl here.

Reply:
I think this is valuable information for future changes and for any new members joining the project. Feel free to include it in the docs wherever you think it fits.